src/common.jl

```diff
@@ -91,6 +91,9 @@ end
 #
 ###########################################################
 
+_sqrt(a::AbstractArray{<:Integer}) = sqrt.(a)
```

Comment:
This one is only used for intermediate results, not for in-place return values.

Reply:
Maybe it would be useful to change `sumsq_percol` instead, so that the resulting array contains floating point numbers if one intends to compute the square root of these numbers later on? Then one could avoid to allocate these additional arrays here.